Week2/todo_app/program.js

```diff
@@ -22,38 +23,91 @@ class TodoModel {
     }
 
     create(description, callback) {
-        // Write code and query to create a new TODO item
+        const queryString = `INSERT INTO todo_items(text, is_completed, user_id) VALUES ('${description}', 0, ${args2})`;
+        this.dbConnection.query(queryString, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results)
+            }
+        });
     }
 
     update(id, description, callback) {
-        // Write code and query to update and existing TODO item
+        const queryString = `UPDATE todo_items SET text = '${description}' WHERE id = ${id}`
+        this.dbConnection.query(queryString, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results)
+            }
+        })
     }
 
     delete(id, callback) {
-        // Write code and query to delete an existing TODO item
+        const queryString = ` DELETE FROM todo_items WHERE id = ${id}`;
+        this.dbConnection.query(queryString, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results)
+            }
+        })
     }
 
     tagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query add a tag to a TODO item
+        const queryString = `INSERT INTO todo_item_tag(todo_item_id, tag_id) VALUES (${todoItemId}, ${tagId})`;
+        this.dbConnection.query(queryString, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            } else {
+                callback(null, results)
+            }
+        })
     }
-        
+
     untagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query remove a tag from a TODO item
+        const queryString = `DELETE FROM todo_item_tag WHERE todo_item_id = ${todoItemId}`;
```

Comment:
This deletes all the tags for a given item. You should be able to remove just one tag from an item, but leave the other tags still attached.